src/test/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParamsTest.java

```diff
@@ -63,6 +66,113 @@ public void getsFilterTemplateWhenValidToDate() {
+        assertThat(transactionSearchParams.getFilterTemplatesWithParentTransactionSearch().get(0), is(" t.created_date < :to_date"));
+        assertThat(transactionSearchParams.getQueryMap().get("to_date").toString(), is("2018-09-22T10:14:16.067Z"));
+    }
+
+    @Test
+    public void getFilterTemplateWithParentTxnSearch_shouldReturnFilterWithGatewayAccountId() {
+        transactionSearchParams.setAccountId("1");
+        assertThat(transactionSearchParams.getFilterTemplatesWithParentTransactionSearch().get(0), is(" t.gateway_account_id = :account_id"));
+        assertThat(transactionSearchParams.getQueryMap().get("account_id"), is("1"));
+    }
+
+    @Test
+    public void getFilterTemplateWithParentTxnSearch_shouldReturnFilterWithType() {
+        transactionSearchParams.setTransactionType(TransactionType.PAYMENT);
+        assertThat(transactionSearchParams.getFilterTemplatesWithParentTransactionSearch().get(0), is(" t.type = :transaction_type::transaction_type"));
+        assertThat(transactionSearchParams.getQueryMap().get("transaction_type"), is(TransactionType.PAYMENT));
+    }
+
+    @Test
+    public void getFilterTemplateWithParentTxnSearch_shouldReturnFilterWithFirstDigitsCardNumber() {
+        transactionSearchParams.setFirstDigitsCardNumber("1234");
+        assertThat(transactionSearchParams.getFilterTemplatesWithParentTransactionSearch().get(0), is(" t.first_digits_card_number = :first_digits_card_number"));
+        assertThat(transactionSearchParams.getQueryMap().get("first_digits_card_number"), is("1234"));
+    }
+
+    @Test
+    public void getFilterTemplateWithParentTxnSearch_shouldReturnFilterWithState() {
+        transactionSearchParams.setState("success");
+        assertThat(transactionSearchParams.getFilterTemplatesWithParentTransactionSearch().get(0), is(" t.state IN (<state>)"));
+        assertThat(((List) transactionSearchParams.getQueryMap().get("state")).get(0), is("SUCCESS"));
+    }
+
+    @Test
+    public void getFilterTemplateWithParentTxnSearch_shouldReturnFilterWithPaymentState() {
+        transactionSearchParams.setPaymentStates(new CommaDelimitedSetParameter("success"));
+        assertThat(transactionSearchParams.getFilterTemplatesWithParentTransactionSearch().get(0), is("( (t.state IN (<payment_states>) AND t.type =  'PAYMENT'::transaction_type))"));
+        assertThat(((List) transactionSearchParams.getQueryMap().get("payment_states")).get(0), is("SUCCESS"));
+    }
+
+    @Test
+    public void getFilterTemplateWithParentTxnSearch_shouldReturnFilterWithRefundState() {
+        transactionSearchParams.setRefundStates(new CommaDelimitedSetParameter("success"));
+        assertThat(transactionSearchParams.getFilterTemplatesWithParentTransactionSearch().get(0), is("( (t.state IN (<refund_states>) AND t.type =  'REFUND'::transaction_type))"));
+        assertThat(((List) transactionSearchParams.getQueryMap().get("refund_states")).get(0), is("SUCCESS"));
+    }
+
```

Comment:
I wonder whether we could parametrise these tests somehow.
They're growing pretty fast.

Reply:
Not sure if it is possible, they are quite different.